### PR TITLE
[Incremental] Another attempt to get the driver to honor -driver-use-frontend-path

### PR DESCRIPTION
**Pull request by davidungar**
This one only subverts the frontend target check if the new frontend path is a Python script. That way, the option still should work with swiftpm. Don't merge until the swiftpm unit tests pass with this.

### CODE REVIEW

**Comment by davidungar** [comment]
@artemcm What do you think about this approach? @DougGregor Do you want to weigh in? Is there a better alternative?

File: Sources/SwiftDriver/Driver/Driver.swift [hunk 1]
```diff
@@ -2062,23 +2062,24 @@ extension Driver {
...
+        let frontendPath = try AbsolutePath(validating: frontendPathString)
+        toolchain.overrideToolPath(.swiftCompiler, path: frontendPath)
         swiftCompilerPrefixArgs = frontendCommandLine
+        hasFrontendBeenRedirectedForTesting = FileType.isFrontendExtensionForTesting(frontendPath.extension)
       }
     } else {
       swiftCompilerPrefixArgs = []
-    }
-    var hasFrontendBeenRedirectedForTesting: Bool {
-      return !swiftCompilerPrefixArgs.isEmpty
```
Comment (artemcm) [inline Sources/SwiftDriver/Driver/Driver.swift:2081]: I find this to be quite confusing and the new approach is much more concrete so I think it is an improvement.

**Review by artemcm — APPROVED** [review]

File: Sources/SwiftDriver/Driver/Driver.swift (hunk 1 above)
Comment (davidungar) [reply Sources/SwiftDriver/Driver/Driver.swift:2081]: Thank you, @artem. Glad you think it's better.

**Comment by davidungar** [comment]
@swift-ci please test

**Pull request merged**